cartographer/mapping/connected_components.cc

```diff
@@ -92,6 +92,7 @@ std::vector<std::vector<int>> ConnectedComponents::Components() {
 }
 
 std::vector<int> ConnectedComponents::GetComponent(const int trajectory_id) {
+  common::MutexLocker locker(&lock_);
```

Comment:
Did you verify this change? I would expect that this does not work as this function gets called from TrajectoryConnectivityState::Connect(...) <= SparsePoseGraph::UpdateTrajectoryConnectivity(...) <= SparsePoseGraph::HandleWorkQueue() which already takes the lock. So I would expect the line above to fail.

Reply:
Thanks for the review! Sorry to reply so late. I verified this change, I think they lock different lockers(their own member lock_), I have to add this because GetComponent calls the FindSet which requires a lock, otherwise the build will fail.

Reply:
Sorry for the delay. I lost sight of this.